staging/src/k8s.io/dynamic-resource-allocation/resourceclaim/resourceclaim.go

```diff
@@ -106,3 +108,37 @@ func CanBeReserved(claim *resourceapi.ResourceClaim) bool {
 	// Currently no restrictions on sharing...
 	return true
 }
+
+// BaseRequestRef returns the request name if the reference is to a top-level
+// request and the name of the parent request if the reference is to a subrequest.
+func BaseRequestRef(requestRef string) string {
+	segments := strings.Split(requestRef, "/")
+	return segments[0]
+}
+
+// ConfigForResult returns the configs that are applicable to device
+// allocated for the provided result.
+func ConfigForResult(deviceConfigurations []resourceapi.DeviceAllocationConfiguration, result resourceapi.DeviceRequestAllocationResult, driver string) []resourceapi.DeviceAllocationConfiguration {
```

Comment:
I think we shouldn't filter by driver here.

A driver might support more than just configuration formats defined for itself. For example, a driver `net.example.com` might support a configuration for a generic `net.kubernetes.io` pseudo-driver, with content defined by Kubernetes.

Also, if the driver is asked to apply a configuration that it doesn't support, it should have the ability to raise an error. It can't do that when using the function as it stands because "unknown" configs are silently ignored.

Reply:
good points, that makes sense to me

Reply:
Yeah, I think that makes sense. I've updated the `ConfigForResult` function to not filter out configs based on the driver. So it will be up to the driver itself to determine how to handle those. The check for the `requests` field still apply.

Reply:
Sorry I missed that. Fixed now.